source/yats/include/yats/identifier.h

```diff
@@ -1,10 +1,17 @@
 #pragma once
 #include <algorithm>
 #include <cstdint>
-#include <cstdlib>
+#include <string>
 
 namespace yats
 {
+// Defines or protocol to convert a string to an id.
+// Size of a single character in bits
+const size_t width_of_character = 5;
+// Number of bits used to encode the string length.
+const size_t width_of_length = 4;
+// Id for first special character
+const size_t unique_begin = 26;
```

Comment:
choose a better name for this. Also most of the other functions have bad names. It is a good idea to use a strict naming convention for these function.